File: src/modifiers/saturation.ts
import { createModifier } from './createModifier';

/// SwiftUI `.saturation(_:)`. Adjusts the color saturation of the view.
/// `1` is unchanged, `0` is grayscale, and values `> 1` oversaturate.
export function saturation(value: number) {
  return createModifier('saturation', { value });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/Saturation.swift
import SwiftUI

/// SwiftUI `.saturation(_:)`. Adjusts color saturation; `1` leaves the
/// view unchanged, `0` is grayscale.
enum RNWSaturationModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("saturation") { view, params, _ in
            AnyView(view.saturation(params.double("value") ?? 1))
        }
    }
}
